jericho/jericho.py

```diff
@@ -522,8 +527,29 @@ def get_valid_actions(self, use_object_tree=True):
         interactive_objs  = self._identify_interactive_objects(use_object_tree=use_object_tree)
         best_obj_names    = self._score_object_names(interactive_objs)
         candidate_actions = self.act_gen.generate_actions(best_obj_names)
-        diff2acts         = self._filter_candidate_actions(candidate_actions)
-        valid_actions     = [max(v, key=utl.verb_usage_count) for v in diff2acts.values()]
+        if use_ctypes:
+            state = self.get_state()
+            candidate_str = (";".join(candidate_actions)).encode('utf-8')
+            valid_str = (' '*len(candidate_str)).encode('utf-8')
```

Comment:
In the C function `self.frotz_lib.filter_candidate_actions` a `; ` is appended after each valid action, i.e.
> valid_actions[v_idx++] = ';';

This means `valid_str` length is short by 1 character (in the case where all candidate actions are valid). Here's a quick fix:
> valid_str = (' '*(len(candidate_str) + 1)).encode('utf-8')